tests/test_visualisation.py

```diff
@@ -98,15 +113,52 @@ def test_scatter_plot_all_cohorts(self):
         mapping_table4 = MappingTable(self.mapping_source)
         mapping_table4.add_descriptions(self.data_dictionary_source)
 
-        mapping_table1.joined_mapping_table['embedding'] = self.embeddings1
-        mapping_table2.joined_mapping_table['embedding'] = self.embeddings2
-        mapping_table3.joined_mapping_table['embedding'] = self.embeddings3
-        mapping_table4.joined_mapping_table['embedding'] = self.embeddings4
-        scatter_plot_all_cohorts([mapping_table1, mapping_table2], [mapping_table3, mapping_table4],
-                                 ["A1", "A2"], ["B1", "B2"], store_html=False)
+        mapping_table1.joined_mapping_table["embedding"] = self.embeddings1
+        mapping_table2.joined_mapping_table["embedding"] = self.embeddings2
+        mapping_table3.joined_mapping_table["embedding"] = self.embeddings3
+        mapping_table4.joined_mapping_table["embedding"] = self.embeddings4
+        scatter_plot_all_cohorts(
+            [mapping_table1, mapping_table2],
+            [mapping_table3, mapping_table4],
+            ["A1", "A2"],
+            ["B1", "B2"],
+            store_html=False,
+        )
 
     def test_enrichment_plot(self):
         acc_gpt = [0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.0, 1.0, 1.0]
+        acc_mpnet = [0.3, 0.4, 0.5, 0.6, 0.8, 0.9, 1.0, 1.0, 1.0, 1.0]
         acc_fuzzy = [0.2, 0.3, 0.4, 0.5, 0.8, 0.9, 1.0, 1.0, 1.0, 1.0]
         title = "Test"
-        enrichment_plot(acc_gpt, acc_fuzzy, title, save_plot=False)
+        enrichment_plot(acc_gpt, acc_mpnet, acc_fuzzy, title, save_plot=False)
+
+    def test_bar_chart_average_acc_two_distributions(self):
```

Comment:
You did a great job at exception handling exceptions in your implementation -  best practice would here to also test for the opposite case; meaning you check for each exception case in a separate test function if your expected exception is raised for a wrong function input.

Python uses the [assert.AssertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) function.